sys/include/net/sock/ip.h

```diff
@@ -12,7 +12,244 @@
+ * one network device.
+ *
+ * After including header files for the @ref net_af "address families",
+ * @ref net_protnum "protocol numbers" and the @ref net_sock_ip "raw `sock`s"
+ * themselves, we create some buffer space `buf` to store the data received by
+ * the server:
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * #include "net/af.h"
+ * #include "net/protnum.h"
+ * #include "net/sock/ip.h"
+ *
+ * uint8_t buf[128];
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * To be able to listen for incoming packets we want to bind the sock so we need
+ * to set a local end point (even if we just state here, that we just want to
+ * bind it to any IPv6 address).
+ *
+ * We then proceed to create the `sock`. It is bound to `local` and listens for
+ * IPv6 packets with @ref ipv6_hdr_t::nh "next header field"
+ * @ref PROTNUM_IPV6_NONXT. Since we don't need any further configuration we set
+ * the flags to 0. In case of an error we stop the program:
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ *     sock_ip_ep_t local = SOCK_IPV6_EP_ANY;
+ *     sock_ip_t sock;
+ *
+ *     if (sock_ip_create(&sock, &local, NULL, PROTNUM_IPV6_NONXT, 0) < 0) {
+ *         puts("Error creating raw IP sock");
+ *         return 1;
+ *     }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * We then wait indefinitely for an incoming message in `buf` from `remote`.
+ * If we want to timeout this wait period we could alternatively set the
+ * `timeout` parameter of @ref sock_ip_recv() to a value `> 0`.
+ * If an error occurs on receive we just ignore it and continue looping.
+ *
+ * If we receive a message we use its `remote` to reply. Note since the `proto`
+ * was already set during @ref sock_ip_create() we can just leave `proto` for
+ * the @ref sock_ip_send() set to 0 (it is ignored by that function in that case
+ * anyway). In case of an error on send we print an according message:
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ *     while (1) {
+ *         sock_ip_ep_t remote;
+ *         ssize_t res;
+ *
+ *         if ((res = sock_ip_recv(&sock, buf, sizeof(buf), 0, &remote)) >= 0) {
+ *             puts("Received a message");
+ *             if (sock_ip_send(&sock, buf, res, 0, &remote) < 0) {
+ *                 puts("Error sending reply");
+ *             }
+ *         }
+ *     }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * ### A Simple IPv6 Client
+ * There are two kinds of clients. Those that do expect a reply and those who
+ * don't. A client that does not require a reply is very simple to implement in
+ * one line:
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * res = sock_ip_send(NULL, data, data_len, PROTNUM, &remote);
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * With `data` being the data send, `data_len` the length of `data`, `PROTNUM`
+ * the next header number for the send packet and `remote` the remote end point
+ * the packet is to be sent.
+ *
+ * To see some other utilization we look at a more complex example in form of
```

Comment:
I disagree with usage here. I'm showing off the capabilities here, not its usage (which was already shown in the one-liner above)